protocol/src/types/cstring.rs

```diff
@@ -0,0 +1,61 @@
+use std::ffi::CString;
+use std::io::prelude::{Read, Write};
+use {hint, Error, Parcel, Settings};
+
+impl Parcel for CString {
+    const TYPE_NAME: &'static str = "CString";
+
+    fn read_field(
+        read: &mut dyn Read,
+        settings: &Settings,
+        _hints: &mut hint::Hints,
+    ) -> Result<Self, Error> {
+        let mut result = Vec::new();
+        loop {
+            let c: u8 = Parcel::read(read, settings)?;
+            if c == 0x00 {
+                break;
+            }
+            result.push(c);
+        }
+        Ok(CString::new(result)?)
+    }
+
+    fn write_field(
+        &self,
+        write: &mut dyn Write,
+        settings: &Settings,
+        _hints: &mut hint::Hints,
+    ) -> Result<(), Error> {
+        for c in self.clone().into_bytes() {
+            c.write(write, settings)?;
+            if c == 0x00 {
```

Comment:
There are two divergent code paths here where the null terminator will be written by either line 31, or line 36.

I would prefer the null byte always being written by the same statement rather at two possible different points within the function.

*NOTE* `CString::into_bytes` is guaranteed to not have any interior null bytes according to the [documentation](https://doc.rust-lang.org/std/ffi/struct.CString.html#method.into_bytes, therefore the `c == 0x00` can never be reached according to the doc. In this case, this early return must be redundant.

*recommendation*: Remove this early return